### PR TITLE
expanded weights: layer norm faster rule

### PR DESCRIPTION
**Pull request by samdow**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #70141
* #73696
* __->__ #73694

Differential Revision: [D34719299](https://our.internmc.facebook.com/intern/diff/D34719299)

### CODE REVIEW

**Comment by samdow** [comment]
@samdow has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34719299).

**Comment by samdow** [comment]
@samdow has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34719299).

**Comment by samdow** [comment]
@samdow has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34719299).

**Review by zou3519 — APPROVED** [review]
LGTM, one question

File: test/test_expanded_weights.py [hunk 1]
```diff
@@ -242,6 +242,8 @@ def convnet(num_classes):
 
 class TestExpandedWeightModule(TestCase):
     def _do_test(self, module, input):
+        if len(list(module.parameters())) == 0:  # for norms with affine=False
+            raise unittest.SkipTest("affine=False, no params")
```
[the remark this answers is not in the document]
Comment (samdow) [reply test/test_expanded_weights.py]: done!

**Comment by samdow** [comment]
Errors seem to be from commit rebased off of or cuda issues (happening on master rn), going to merge

**Pull request closed without merging**